### ISSUE TITLE
Wifi not working

### ISSUE THREAD

**Issue by Jasjar42** [neagix/wii-linux-ngx#14]
Wifi doesn't work for me. The wifi setup script just reports it couldn't connect. Knowing how WPA2 doesn't work I made sure to change it to WPA.  WEP isn't an option. The wifi works fine for the system menu. I'm using the newest stable release.

**Comment by neagix** [comment]
It works for me. So now you would have to report the iwconfig information of the network you're trying to connect to, otherwise 'Wifi not working' is too generic (as for other users it's working).

Also, errors in dmesg

**Comment by Jasjar42** [comment]
Sorry for not replying. I can't get the iwconfig information since I don't have a linux device connected to wifi. I have the ifconfig information though. I have the dmesg as well
[dmesg.txt](https://github.com/neagix/wii-linux-ngx/files/1384446/dmesg.txt)
![img_20171014_163812](https://user-images.githubusercontent.com/19740686/31573216-1825c668-b103-11e7-886e-1347ccc75c89.jpg)
![ifconfig](https://user-images.githubusercontent.com/19740686/31573222-27bfa31e-b103-11e7-8874-dababeec8b7b.png)

**Comment by Brawl345** [comment]
Fiddled around with this a bit yesterday and can confirm that the whiite-ez-wifi-config script creates an invalid config - the interface is wlan1, not wlan0 (which can be confirmed with "ip link"). Run the script again and edit the /etc/network/interfaces and replace every "wlan0" with "wlan1", save and bring up the network with "ifup wlan1".
You can also confirm this by looking into the dmesg file:
```
[    3.985971] systemd-udevd[591]: renamed network interface wlan0 to wlan1
```

**Comment by Jasjar42** [comment]
Thank you. Wifi now works for me.

**Comment by neagix** [comment]
Does this happen only to users that also have a ethernet dongle? Or it's unrelated?

**Comment by Jasjar42** [comment]
It's unrelated, if I had one I would be using it.

**Comment by Brawl345** [comment]
I also don't have an Ethernet dongle.

**Comment by SpazmasterYT** [comment]
Exact same senario for me, fixed it by changing wlan0 to wlan1 in the script

**Comment by immolo** [comment]
This issue is still happening without an ethernet dongle, is everyone using wlan1 or is it just some models?

**Comment by MrQbasic** [comment]
I have a question: I don't want to so change the encryption on my main network. So can I instead use my phone to host a networking with my mobile data (I don't know the English word) wich is open because I can't change the encryption type on my phone. So basicly how do I confic my interfaces file so that it works with open networks?

**Comment by tesses50** [comment]
i had to edit whiite-linux-ez-config to use wlan1 instead of wlan0

Is it just that wifi is slow (arround ~400KB when torrenting) we got 25 Mb plan (yes I know the difference between megabit and megabyte) or is it the wii that is slow too (I know we arent getting exactly 25Mb but still) (I know this part is unrelated to the github issue)

**Comment by Brawl345** [comment]
> is it the wii that is slow too

yes, the Wii's wifi is horrible for today's standards. Compare it to the Old3DS which released 2011 and only can do about 700 KB/s.

**Comment by tesses50** [comment]
would the wii be that slow with ethernet adapter (I tried one and it only worked once (then would need to reboot to use it again)), I use my wii for torrenting, downloading youtube videos (must be an ocd thing or something, I know I have autism)

**Comment by immolo** [comment]
> would the wii be that slow with ethernet adapter (I tried one and it only worked once (then would need to reboot to use it again)), I use my wii for torrenting, downloading youtube videos (must be an ocd thing or something, I know I have autism)

I find the adapter slower than using WiFi but I've not tried using a newer kernel yet to see if that helps.

**Comment by naruse** [comment]
> Fiddled around with this a bit yesterday and can confirm that the whiite-ez-wifi-config script creates an invalid config - the interface is wlan1, not wlan0 (which can be confirmed with "ip link"). Run the script again and edit the /etc/network/interfaces and replace every "wlan0" with "wlan1", save and bring up the network with "ifup wlan1". You can also confirm this by looking into the dmesg file:
> 
> ```
> [    3.985971] systemd-udevd[591]: renamed network interface wlan0 to wlan1
> ```

Necro-bumping this but thanks!, indeed the interface had to be renamed to wlan1 and it works perfectly! thanks!

So just change where it says `wlan0` in /etc/network/interfaces to `wlan1`

**Comment by MrQbasic** [comment]
Thx so much

**Comment by EldosHD** [comment]
Hate to necropost, but i already did the fix with 'wlan1' instead of 'wlan0' and the wii does now connect to my router. But it can't do anything wifi related. 'apt-get update', 'wget <vaild url>' ... etc. doesnt work. Any ideads whats causing this?

**Comment by naruse** [comment]
> Hate to necropost, but i already did the fix with 'wlan1' instead of 'wlan0' and the wii does now connect to my router. But it can't do anything wifi related. 'apt-get update', 'wget ' ... etc. doesnt work. Any ideads whats causing this?

the problem is that for today (May 2023) that distro is _really_ old so apt-get keys are expired so it will just not work; wget should work though and basic internet connectivity but package management will not work.

The bigger problem is that the wii is a PPC 32 bit architecture and pretty much the support is gone from the distros that used to support it.

**Comment by EldosHD** [comment]
Thx for the quick answer lol.
I guess I just stop bothering then... I just need to figure out what to do with my second wii now...

**Comment by juansebastianUTech** [comment]
> Thx for the quick answer lol. I guess I just stop bothering then... I just need to figure out what to do with my second wii now...

there was an attempt to make FreeBSD (or OpenBSD? cant remember) to boot linux in the wii; I havent checked it out with more detail, but I think that would be the last resource. if you find anything useful to do with it let us know!

**Comment by MrQbasic** [comment]
I think there is Linux distro out there focusing on ppc support. It is maintained by one guy though so idk how well it is doing. Also you probably would need to make the kernel modules compatible with the distro. Idk the detail as I'm not a distro dev.
